core/trino-main/src/main/java/io/trino/connector/ConnectorManager.java

```diff
@@ -337,12 +337,14 @@ private synchronized void removeConnectorInternal(CatalogName catalogName)
         MaterializedConnector materializedConnector = connectors.remove(catalogName);
         if (materializedConnector != null) {
             Connector connector = materializedConnector.getConnector();
-            try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(connector.getClass().getClassLoader())) {
-                connector.shutdown();
-            }
-            catch (Throwable t) {
-                log.error(t, "Error shutting down connector: %s", catalogName);
-            }
+            withClassLoader(connector.getClass().getClassLoader(), () -> {
```

Comment:
```suggestion
            withClassLoader(connector.getClass(), () -> {
```

Reply:
thanks, fixed this and bunch of others in https://github.com/trinodb/trino/pull/8507/commits/b01458ea6fe70be739cf138df909969b10e08210